src/library/scala/collection/mutable/BufferLike.scala

```diff
@@ -105,15 +105,18 @@ trait BufferLike[A, +This <: BufferLike[A, This] with Buffer[A]]
     */
   def remove(n: Int): A
 
-  /** Removes a number of elements from a given index position.
+  /** Removes a number of elements from a given index position.  Subclasses of `BufferLike`
+   *  will typically override this method to provide better performance than `count`
+   *  successive calls to single-element `remove`.
    *
    *  @param n  the index which refers to the first element to remove.
    *  @param count  the number of elements to remove.
    *  @throws   IndexOutOfBoundsException if the index `n` is not in the valid range
-   *            `0 <= n <= length - count`.
+   *            `0 <= n <= length - count` (with `count > 0`).
```

Comment:
I am not exactly sure what I think about it either, but I am almost positive that nobody thought carefully about what exactly the error cases should be for each method.  So, I propose: if you ask for nothing, and do not need to check anything to know that it is nothing, then nothing happens and it happens safely.  (I'd call this the safe-no-op property.)  Scala's collections tend to lean towards that property.